fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java

```diff
@@ -123,8 +126,9 @@ public void applyVisibleLog(TransactionState txnState, TableCommitInfo commitInf
                         long lastFailedVersion = replica.getLastFailedVersion();
                         long newVersion = version;
                         long lastSucessVersion = replica.getLastSuccessVersion();
-                        if (!txnState.tabletCommitInfosContainsReplica(tablet.getId(), replica.getBackendId(),
-                                replica.getState())
+                        if ((!txnState.tabletCommitInfosContainsReplica(tablet.getId(), replica.getBackendId(),
+                                replica.getState()) && !txnState.
+                                checkTransactionDependencyReplicasNotCommited((LocalTablet) tablet, quorumReplicaNum))
```

Comment:
in this case, replicaC does not have the new version rowset but the version is updated to newVersion, this is not right?